Assets/Simon/Scripts/ColorButton.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class ColorButton : MonoBehaviour
+{
+    [SerializeField] private Colors color;
+
+    [SerializeField] private AudioClip _audioClip;
+
+    public AudioClip AudioClip
+    {
+        get
+        {

```

Comment:
Done. I just want to know why this is wrong. I did what I did to so that the audio clip can't be changed at runtime from another script. This is why there was no set for AudioClip. Is this correct?

Reply:
```csharp
public AudioClip AudioClip => _audioClip;
```
This is readonly. There is no Set defined on the property.

It's just the modern C# way of doing things (called an 'expression-bodied member'): https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/expression-bodied-members. 


It is equivalent to:

```csharp
public AudioClip AudioClip { get { return _audioClip; } }
```

You don't actually need the separate serialised field too btw. You can do: 
```csharp 
[field: SerializeField] 
public AudioClip { get; private set; }
```
Which serialises the backing field on the property for you automatically.

Reply:
Ok got it.